modules/Rock.cmake

```diff
@@ -1350,6 +1366,39 @@ function(rock_gtest TARGET_NAME)
 
     rock_setup_gtest_test(${TARGET_NAME} ${GMOCK_DIR} ${GTEST_DIR})
     rock_add_test(${TARGET_NAME} "${__rock_test_parameters}")
+    list(APPEND ALL_TEST_TARGETS "test-${TARGET_NAME}-cxx")
+    set(ALL_TEST_TARGETS "${ALL_TEST_TARGETS}" CACHE INTERNAL "ALL_TEST_TARGETS")
+endfunction()
+
+function(rock_cxx_coverage_report PROJECT_NAME)
+    message(STATUS "Generating report for ${PROJECT_NAME} package")
+    find_program(
+        gcovr_exec NAMES
+        ${ROCK_GCOVR_EXECUTABLE} gcovr
+    )
+    if (NOT gcovr_exec)
+        message(FATAL_ERROR "Could not find an executable for gcovr.")
+    endif()
+
+    if(${ROCK_COVERAGE_REPORT_PATH})
+        message(STATUS "Setting an explicit config path coverage report")
+        set(gcovr_config_option --filter ${PROJECT_SOURCE_DIR}/src/ --html -o ${ROCK_COVERAGE_REPORT_PATH}/coverage.html)
+    else()
+        message(STATUS "Default coverage path set to /build")
+        set(gcovr_config_option --filter ${PROJECT_SOURCE_DIR}/src/ --html -o ${PROJECT_BINARY_DIR}/coverage.html)
+    endif()
+
+    add_test(
+        NAME
+        ${PROJECT_NAME}_report_generation
+        COMMAND
+        ${gcovr_exec}
+        ${gcovr_config_option}
+        ${gcovr_extra_option}
+    )
+    foreach(target ${ALL_TEST_TARGETS})
```

Comment:
Use ARGN instead of ALL_TEST_TARGETS here, and explicitly pass the targets to the call.

Reply:
This becomes deprecated by the top comment doesn't it? Changes made